build-caching/src/main/java/com/vertispan/j2cl/build/TaskScheduler.java

```diff
@@ -199,33 +206,35 @@ private void scheduleMoreWork(DiskCache.CacheResult cacheResult) {
     }
 
     private void executeFinalTask(CollectedTaskInputs taskDetails, DiskCache.CacheResult cacheResult) throws Exception {
-        System.out.println("starting final task " + taskDetails.getProject().getKey() + " " + taskDetails.getTaskFactory().getOutputType());
+        buildLog.info("starting final task " + taskDetails.getDebugName());
         long start = System.currentTimeMillis();
         try {
             ((TaskFactory.FinalOutputTask) taskDetails.getTask()).finish(new TaskOutput(cacheResult.outputDir()));
-            System.out.println(taskDetails.getProject().getKey() + " final task " + taskDetails.getTaskFactory().getOutputType() + " finished in " + (System.currentTimeMillis() - start) + "ms");
+            buildLog.info("Finished " + taskDetails.getDebugName() + " in " + (System.currentTimeMillis() - start) + "ms");
         } catch (Throwable t) {
-            System.out.println(taskDetails.getProject().getKey() + " final task " + taskDetails.getTaskFactory().getOutputType() + " failed in " + (System.currentTimeMillis() - start) + "ms");
+            buildLog.info("FAILED   " + taskDetails.getDebugName() + " in " + (System.currentTimeMillis() - start) + "ms");
```

Comment:
this is probably an ERROR/WARN rather than INFO with a "FAILED" string?

Reply:
...more likely a DEBUG/TRACE, depending on how long the task takes, but we can deal with that later.

Reply:
Maven has no "trace" level, so debug could be used. But since we throw the Throwable anyway, error seems fitting.

Reply:
Maven doesn't, but other potential consumers of the library code might...